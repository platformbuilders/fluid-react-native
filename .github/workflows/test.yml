name: test

on:
  pull_request:
    branches:
      - master

jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - uses: actions/setup-node@v1
        with:
          node-version: '14.x'
      # - name: Set NPM Variables
      #   run: echo "$PACKAGES_NPMRC" > .npmrc
      #   env:
      #     PACKAGES_NPMRC: ${{ secrets.PACKAGES_NPMRC }}
      - name: Install packages
        run: yarn install
      - name: Test
        run: yarn test --coverage
