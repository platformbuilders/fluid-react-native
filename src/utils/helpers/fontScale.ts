import { RFValue } from 'react-native-responsive-fontsize';

export const scaledFontSize = (
  size: number | string,
  baselineFactor?: 280,
): number => {
  const value = typeof size === 'string' ? parseInt(size, 10) : size;
  return RFValue(value, baselineFactor);
};
