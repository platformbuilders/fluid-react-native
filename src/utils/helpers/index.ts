export {
  currencyParser,
  parseToThousands,
  removeWhiteSpaces,
  toOnlyNumbers,
  getTheme,
  ifStyle,
  switchStyle,
} from '@platformbuilders/helpers';
export * from './generateHaptic';
export * from './getByTypographyVariant';
export * from './fontScale';
export * from './shadow';
