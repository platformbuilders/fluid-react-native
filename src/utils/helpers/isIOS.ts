import { Platform } from 'react-native';

const IOS_PLATFORM = 'ios';

export const isIOS = (): boolean => Platform.OS === IOS_PLATFORM;
