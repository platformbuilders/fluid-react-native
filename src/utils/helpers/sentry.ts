import { Sentry } from '.';

/**
 * Generic function to be used with Sentry. Under the hood, it's called the `addBreadcrumb` method
 * from Sentry API, which adds logs about user's actions prior to an error or crash.
 *
 * Read [this research doc](https://s3.us-west-2.amazonaws.com/secure.notion-static.com/89ba7171-90e3-4e5b-9856-fe36be0bed14/Sentry_-_Sentry_API_study_%28addBreadcrumb%29.pdf?X-Amz-Algorithm=AWS4-HMAC-SHA256&X-Amz-Content-Sha256=UNSIGNED-PAYLOAD&X-Amz-Credential=AKIAT73L2G45EIPT3X45%2F20230119%2Fus-west-2%2Fs3%2Faws4_request&X-Amz-Date=20230119T180359Z&X-Amz-Expires=86400&X-Amz-Signature=b0395d3285cebaa0562ac64b30ed50a3803928181b7a40428c81dfd000442f03&X-Amz-SignedHeaders=host&response-content-disposition=filename%3D%22Sentry_-_Sentry_API_study_%28addBreadcrumb%29.pdf%22&x-id=GetObject)
 * to get more information about it's usage.
 * @param param0
 * @returns
 */
export const handleSentryTraceability = ({
  useSentryTraceability = false,
  category,
  data,
  event_id,
  level = 'info',
  message,
  timestamp,
  type,
}: HandleSentryTraceabilityProps) => {
  if (!useSentryTraceability) return;

  Sentry.addBreadcrumb({
    category,
    data,
    event_id,
    level,
    message,
    timestamp,
    type,
  });
};
