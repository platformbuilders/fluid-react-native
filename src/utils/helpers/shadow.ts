import { moderateScale } from 'react-native-size-matters';

const shadowRadius = `${moderateScale(4)}px`;

export const getShadow = (): string => `
  shadow-offset: 0px 4px;
  shadow-opacity: 0.2;
  shadow-radius: ${shadowRadius};
  elevation: 3;
`;
