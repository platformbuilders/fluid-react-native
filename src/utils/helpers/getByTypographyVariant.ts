import { getTheme } from '@platformbuilders/helpers';
import { scaledFontSize } from './fontScale';

export const getFontSize = (props: any): number => {
  const fontSize = getTheme(`${props.variant}.fontSize`)(props) || 16;
  return scaledFontSize(fontSize);
};

export const getLineHeight = (props: any): number => {
  const lineHeight = getTheme(`${props.variant}.lineHeight`)(props) || 22;
  return scaledFontSize(lineHeight);
};
