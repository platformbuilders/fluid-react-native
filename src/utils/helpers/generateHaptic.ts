import Haptic, { HapticFeedbackType } from 'react-native-haptic';
import { isIOS } from './isIOS';

export const generateHaptic = (haptic: HapticFeedbackType = 'impact'): void => {
  if (isIOS()) {
    Haptic.generate(haptic);
  }
};
