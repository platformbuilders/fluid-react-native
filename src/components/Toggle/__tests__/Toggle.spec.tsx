import React from 'react';
import { fireEvent, render } from 'react-native-testing-library';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components/native';
import { theme } from '../../../test/helpers';
import ToggleButton from '../index';

const onPressEvent = jest.fn();

describe('<ToggleButton />', () => {
  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('should render toggle component', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <ToggleButton value={true} onValueChange={() => {}} />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should trigger onValueChange function when toggle is pressed', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ToggleButton value={false} onValueChange={onPressEvent} />
      </ThemeProvider>,
    );

    const component = getByTestId('toggle_button');

    fireEvent(component, 'onValueChange', true);

    expect(onPressEvent).toHaveBeenCalledTimes(1);
  });

  it('should render component with custom thumb color', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <ToggleButton
          value={true}
          onValueChange={() => {}}
          thumbColor={{ trueColor: 'red', falseColor: 'blue' }}
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render component with custom track color', () => {
    const wrapper = renderer.create(
      <ThemeProvider theme={theme}>
        <ToggleButton
          value={true}
          onValueChange={() => {}}
          trackColor={{ trueColor: 'red', falseColor: 'blue' }}
        />
      </ThemeProvider>,
    );

    expect(wrapper.toJSON()).toMatchSnapshot();
  });

  it('should render component disabled if disable prop is passed', () => {
    const { getByTestId } = render(
      <ThemeProvider theme={theme}>
        <ToggleButton value={true} onValueChange={() => {}} isDisabled />
      </ThemeProvider>,
    );

    const component = getByTestId('toggle_button');

    expect(component).toHaveProperty(
      ['_fiber', 'memoizedProps', 'disabled'],
      true,
    );
  });
});
