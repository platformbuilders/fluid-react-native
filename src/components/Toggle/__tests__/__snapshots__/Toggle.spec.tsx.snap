// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ToggleButton /> should render component with custom thumb color 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#007AFF80"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Array [
          Object {},
        ],
        Object {
          "backgroundColor": "#0000004D",
          "borderRadius": 16,
        },
      ],
    ]
  }
  testID="toggle_button"
  thumbColorProps={
    Object {
      "falseColor": "blue",
      "trueColor": "red",
    }
  }
  thumbTintColor="red"
  tintColor="#0000004D"
  value={true}
/>
`;

exports[`<ToggleButton /> should render component with custom track color 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="red"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Array [
          Object {},
        ],
        Object {
          "backgroundColor": "blue",
          "borderRadius": 16,
        },
      ],
    ]
  }
  testID="toggle_button"
  thumbTintColor="#007AFF"
  tintColor="blue"
  trackColorProps={
    Object {
      "falseColor": "blue",
      "trueColor": "red",
    }
  }
  value={true}
/>
`;

exports[`<ToggleButton /> should render toggle component 1`] = `
<RCTSwitch
  accessibilityRole="switch"
  disabled={false}
  onChange={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  onTintColor="#007AFF80"
  style={
    Array [
      Object {
        "height": 31,
        "width": 51,
      },
      Array [
        Array [
          Object {},
        ],
        Object {
          "backgroundColor": "#0000004D",
          "borderRadius": 16,
        },
      ],
    ]
  }
  testID="toggle_button"
  thumbTintColor="#007AFF"
  tintColor="#0000004D"
  value={true}
/>
`;
